src/QueryConditionGenerator.php

```diff
@@ -81,40 +105,79 @@ private function processGroup(ValuesGroup $group): array
         // FIXME Objects need type formatter. `elastic_search_value_transformer` (use extensions to configure types)
 
         foreach ($group->getFields() as $fieldName => $valuesBag) {
+            $propertyName = $this->mappings[$fieldName]->propertyName;
+            $field = $this->fieldSet->get($fieldName);
+
+            // TODO: hack, review and fix
+            $fieldType = $field->getType()->getInnerType();
+            $dateTransformer = new DateTimeToStringTransformer(null, 'UTC', 'Y-m-d');
+
             if ($valuesBag->hasSimpleValues()) {
-                $bool[$includingType][]['terms'] = [$this->mappings[$fieldName]->indexName => array_values($valuesBag->getSimpleValues())];
+                if ($fieldType instanceof DateType) {
+                    // date lookups behave like a range even for an exact value in Elasticsearch
+                    // https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-range-query.html#ranges-on-dates
+                    $values = array_values($valuesBag->getSimpleValues());
+                    foreach ($values as $idx => $value) {
+                        $valuesBag->removeSimpleValue($idx);
+                        $value = $dateTransformer->transform($value);
+                        $range = new Range($value, $value);
+                        $valuesBag->add($range);
+                    }
+                } else {
+                    $bool[$includingType][]['terms'] = [$propertyName => array_values($valuesBag->getSimpleValues())];
+                }
             }
 
             if ($valuesBag->hasExcludedSimpleValues()) {
-                $bool['must_not'][]['terms'] = [$this->mappings[$fieldName]->indexName => array_values($valuesBag->getExcludedSimpleValues())];
+                $bool['must_not'][]['terms'] = [$propertyName => array_values($valuesBag->getExcludedSimpleValues())];
             }
 
-            /** @var Range $range */
-            foreach ($valuesBag->get(Range::class) as $range) {
-                $rangeParams = [
-                    $range->isLowerInclusive() ? 'lte' : 'lt' => $range->getLower(),
-                    $range->isUpperInclusive() ? 'gte' : 'gt' => $range->getUpper(),
-                ];
+            $isId = self::PROPERTY_ID === $propertyName;
 
-                $bool[$includingType][] = [$this->mappings[$fieldName]->indexName => $rangeParams];
-            }
+            if (false === $isId) {
```

Comment:
This check needs to be done sooner as simple values can also apply to the id, no?

Reply:
Yes, I'll be moving the whole thing.